src/config/input.ts

```diff
@@ -0,0 +1,79 @@
+import {InstallablePhpVersionType} from "./php";
+import {ComposerDependency, IgnorePhpPlatformRequirements} from "./app";
+
+export interface JobToExcludeFromFile {
+    name: string;
+}
+
+export interface ConfigurationFromFile {
+    extensions?: string[];
+    ini?: string[];
+    ignore_php_platform_requirements?: IgnorePhpPlatformRequirements;
+    stablePHP?: string;
+    additional_composer_arguments?: string[];
+}
+
+export interface JobExclusionsFromFile {
+    exclude?: [JobToExcludeFromFile, ...JobToExcludeFromFile[]];
+}
+
+export interface AdditionalChecksConfigurationFromFile extends ConfigurationFromFile, JobExclusionsFromFile {
+    additional_checks?: [JobFromFile, ...JobFromFile[]];
+}
+
+export interface ExplicitChecksConfigurationFromFile extends ConfigurationFromFile {
+    checks?: [JobFromFile, ...JobFromFile[]];
+}
+
+export function isAdditionalChecksConfiguration(config: ConfigurationFromFile): config is AdditionalChecksConfigurationFromFile {
+    return ((config as AdditionalChecksConfigurationFromFile).additional_checks ?? []).length > 0;
+}
+
+export function isExplicitChecksConfiguration(config: ConfigurationFromFile): config is ExplicitChecksConfigurationFromFile {
+    return ((config as ExplicitChecksConfigurationFromFile).checks ?? []).length > 0;
+}
+
+export function isConfigurationContainingJobExclusions(config: ConfigurationFromFile): config is JobExclusionsFromFile&ConfigurationFromFile {
+    return ((config as JobExclusionsFromFile).exclude ?? []).length > 0;
+}
+
+export const LOWEST_ALIAS = '@lowest' as const;
+export const LATEST_ALIAS = '@latest' as const;
+export const WILDCARD_ALIAS = '*' as const;
+
+export type AnyPhpVersionType = typeof WILDCARD_ALIAS;
+export function isAnyPhpVersionType(version: string): version is AnyPhpVersionType {
+    return version === WILDCARD_ALIAS;
+}
+export type LowestPhpVersionType = typeof LOWEST_ALIAS;
+export function isLowestPhpVersionType(version: string): version is LowestPhpVersionType {
+    return version === LOWEST_ALIAS;
+}
+export type LatestPhpVersionType = typeof LATEST_ALIAS;
+export function isLatestPhpVersionType(version: string): version is LatestPhpVersionType {
+    return version === LATEST_ALIAS;
+}
+
+export interface JobFromFile {
+    name: string;
+    job: JobDefinitionFromFile;
+    operatingSystem?: string;
+    action?: string;
+}
+
+export interface JobDefinitionFromFile {
+    php?: InstallablePhpVersionType | AnyPhpVersionType | LowestPhpVersionType | LatestPhpVersionType;
+    extensions?: string[];
+    ini?: string[];
+    dependencies?: AnyComposerDependency | ComposerDependency;
```

Comment:
Suggestion, `depedencyVersionRestraint` or `dependencyRestraint` as noted earlier.

Reply:
There is no way that we change input config. I do not want to introduce BC breaks just because we are discussing namings.
Input JSON and output JSON is not arguable. We either have to life with it or start deprecating stuff. But the benefit in this case is not really huge, is it?

Reply:
Completely fair, I wasn't thinking about this being used outside of this project.